### PR TITLE
diff: add enable option

### PR DESCRIPTION
**Pull request by zowoq**

### CODE REVIEW

File: nixos/common/upgrade-diff.nix [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
-        echo "---"
-      fi
-    '';
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  options.srvos.diff = {
```
Comment (Mic92) [inline nixos/common/upgrade-diff.nix:9]:
```suggestion
  options.srvos.upgrade-diff = {
```

File: nixos/common/upgrade-diff.nix [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+    enable = lib.mkEnableOption "upgrade diff" // {
+      default = true;
+    };
+  };
+  config = lib.mkIf config.srvos.diff.enable {
```
Comment (Mic92) [inline nixos/common/upgrade-diff.nix:14]:
```suggestion
  config = lib.mkIf config.srvos.upgrade-diff.enable {
```

File: nixos/common/upgrade-diff.nix [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
-      fi
-    '';
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  options.srvos.diff = {
+    enable = lib.mkEnableOption "upgrade diff" // {
```
Comment (Mic92) [inline nixos/common/upgrade-diff.nix:10]:
```suggestion
    enable = lib.mkEnableOption "show diff when updating nixos" // {
```

File: nixos/common/upgrade-diff.nix [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+      default = true;
+    };
+  };
+  config = lib.mkIf config.srvos.diff.enable {
+    system.activationScripts.diff = {
```
Comment (Mic92) [inline nixos/common/upgrade-diff.nix:15]:
```suggestion
    system.activationScripts.update-diff = {
```

**Comment by zimbatm** [comment]
:+1: this one could also be upstreamed as well

**Pull request merged**